Feature: Add additional integration GraphQL fns

chasen-bettinger opened the pull request:
- Moved generated.go outside of the jupiterone package and into an internal package
- Added genqlient files so that we can generate additional graphql functions
- Refactored integration code to use genqlient graphql functions
- Added ability to get an integration instance and update an integration instance

Commit:
Feature: Add additional integration GraphQL fns

- Moved generated.go outside of the jupiterone
package and into an internal package
- Added genqlient files so that we can generate
additional graphql functions
- Refactored integration code to use genqlient
graphql functions
- Added ability to get an integration instance
and update an integration instance

Commit:
Chore: Add example of updating an integration

Commit:
Chore: Remove incorrect fatal statements

Commit:
Chore: Updates after precommit make command

zemberdotnet commented on examples/update-integration-instance/example.go:
```diff
@@ -0,0 +1,74 @@
+				DayOfWeek: integrationInstance.IntegrationInstance.PollingIntervalCronExpression.DayOfWeek,
+				Hour:      integrationInstance.IntegrationInstance.PollingIntervalCronExpression.Hour,
+			},
+			Description:     integrationInstance.IntegrationInstance.Name,
```
```suggestion
			Description:     integrationInstance.IntegrationInstance.Description,
```

zemberdotnet commented on internal/graphql/generated.go:
```diff
@@ -0,0 +1,2413 @@
+	SortOrderDesc SortOrder = "DESC"
+)
+
+type UpdateIntegrationInstanceInput struct {
+	Name                          string                                        `json:"name"`
+	SourceIntegrationInstanceId   string                                        `json:"sourceIntegrationInstanceId"`
+	PollingInterval               IntegrationPollingInterval                    `json:"pollingInterval"`
+	PollingIntervalCronExpression IntegrationPollingIntervalCronExpressionInput `json:"pollingIntervalCronExpression"`
+	Description                   string                                        `json:"description"`
+	Config                        map[string]interface{}                        `json:"config"`
+	OffsiteComplete               bool                                          `json:"offsiteComplete"`
+}
```
It doesn't look like the generated code is going to handle optional fields very well. Not all integrations are supposed/want to have `PollingIntervalCronExpressions` and `SourceIntegrationInstanceId`. I'm not sure the API will smoothly handle the empty values Go will serialize.

chasen-bettinger replied:
@zemberdotnet  Great catch! Thanks for looking over my shoulder. Take a peek now, I think it's closer to what we want.

zemberdotnet commented on jupiterone/integrations.go:
```diff
@@ -108,167 +17,33 @@ type DeleteInstanceResult struct {
 
 // ListInstances list the integration instances for the JupiterOne account.
-// ListInstances returns a reference to an IntegrationInstanceResponse which
+// ListInstances returns a reference to an AnIntegrationInstanceResponse which
```
```suggestion
// ListInstances returns a reference to an IntegrationInstanceResponse which
```

Commit:
Chore: Modify UpdateIntegrationInstance

UpdateIntegrationInstance now has an improved
interface that uses optional fields so that
consumers don't upload erroneous values because
of a poor interface.

- Removed sec from Makefile. Need to implement a security
checker on this repository and add it back.
- Added tools.go so that go mod tidy doesn't remove
genqlient from go.mod
- Updated the example to actually work
- Fixed minor NITs

zemberdotnet commented on internal/graphql/generated.go:
```diff
@@ -0,0 +1,2413 @@
+	Description                   string                                        `json:"description"`
+	Config                        map[string]interface{}                        `json:"config"`
+	OffsiteComplete               bool                                          `json:"offsiteComplete"`
+}
```
Might be good to just do the same thing here.

chasen-bettinger replied:
@zemberdotnet done!

Commit:
Chore: Improve CreateIntegrationInstance interface

- Performed the same "optional" operation on the
CreateIntegrationInstanceInput struct as was done
on the UpdateIntegrationInstanceInput struct
